otaclient/app/boot_control/selecter.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022 TIER IV, INC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import platform
+from pathlib import Path
+from typing import Type
+
+from .configs import BootloaderType, cboot_cfg, rpi_boot_cfg
+from .protocol import BootControllerProtocol
+
+from ..configs import config as cfg
+from .. import log_util
+
+logger = log_util.get_logger(
+    __name__, cfg.LOG_LEVEL_TABLE.get(__name__, cfg.DEFAULT_LOG_LEVEL)
+)
+
+
+def detect_bootloader(raise_on_unknown=True) -> BootloaderType:
```

Comment:
Is it possible to detect bootloader type dynamically?
If it is possible, we don't need to ask field integrator guys to add bootloader type in the ecu_info.yaml.

Reply:
@obi-t4 This is the fallback mechanism to detect bootloader if `bootloader` info is not available in `ecu_info.yaml`. 
What I am afraid is that the evidence I use to detect bootloader might become unavailable in the future(i.e., the path to the special file changed, etc.).

Although the breaking changes like the one I mentioned above are very unlikely to happen, but I still prefer explicitly configure `bootloader` type in the `ecu_info.yaml`.

About the backward compatibility, we can still keep the `ecu_info.yaml` unchanged and let this function detect bootloader , but in the future production use(x2 platform), I would like to explicitly set the `bootloader` type in `ecu_info.yaml` when initializing/installing on new devices.

Reply:
> This is the fallback mechanism

I see. The bootloader type handling with ecu_info.yaml is another PR, right? Understood.